### PR TITLE
[Add] Login to Discord and fundamental structures

### PR DESCRIPTION
**Pull request by Colk-tech**
This pull request have these modifications;
- Implemented login to Discord
- Considered fundamental project structure

**I'm brand new to golang, so please go easy on me.**

I would be happy if your review is written from such points of view;
- Structure
- Manner of golang

English and Japanese are permitted to use in reviews of this repository.
Use whichever you like.

Thanks!

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package main
+
+import (
+	setup "./src/setup"
+)
```
Comment (kawaemon) [inline main.go]: Importing with relative path is uncommon now. Please use go modules for future stability.
see: https://qiita.com/momotaro98/items/23fa4356389a7e610acc

TL;DR: create go.mod and import local modules with absolute path (e.g.: `setup github.com/approvers/go-vcnotify/src/setup`

File: src/config/config.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package config
+
```
Comment (kawaemon) [inline src/config/config.go]: on golang, containing source codes in "src" directory is deprecated because you need to specify src module every time you import modules via go modules.

File: src/utils/utils.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package utils
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	config "../config"
```
Comment (mirror-kt) [inline src/utils/utils.go]: 名前付きimportしなくてよさそう

File: src/setup/setup.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/bwmarrin/discordgo"
+
+	handler "../handlers"
```
Comment (mirror-kt) [inline src/setup/setup.go]: 名前付きimportしなくてよさそう

File: src/setup/setup.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	utils "../utils"
+)
+
+
+func Setup() {
```
Comment (mirror-kt) [inline src/setup/setup.go]: SetupがSetup以外の責務まで受け持ってるように見える
Handlerの登録はSetupだとして、シグナルの待ち受け用channelの用意やそれを用いた待ち受けはSetupではないのでは

File: src/utils/utils.go [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+func GetToken() string {
+	token := os.Getenv(config.TokenEnvironmentName)
+
+	if token == "" {
+		displayError := fmt.Sprintf("Error: Unable to get environment veriable: '%s'\n",
+			config.TokenEnvironmentName)
+
+		panic(displayError)
```
Comment (kawaemon) [inline src/utils/utils.go]: using `log.Panicf` is more convenient.
doc: https://golang.org/pkg/log/#Panicf

File: main.go [hunk 7]
```diff
@@ -0,0 +1,9 @@
+package main
+
+import (
+	setup "./src/setup"
```
Comment (mirror-kt) [inline main.go]: 名前付きimportしなくてよさそう

**Review by mirror-kt — CHANGES_REQUESTED** [review]

**Comment by Colk-tech** [comment]
Thank you so much for your reviews!
I learned a lot from that.

I fixeed all the sourcecode along your revies, so would you re-review?

Thanks!

**Review by mirror-kt — APPROVED** [review]

**Review by kawaemon — APPROVED** [review]
seems good now :+1:

File: setup/setup.go [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+	"github.com/approvers/go-vcnotify/handlers"
+	"github.com/approvers/go-vcnotify/utils"
+)
+
+
+func Setup() *discordgo.Session {
+	discord, err := discordgo.New()
+
+	if err != nil {
+		displayError := fmt.Sprintf("Error: Unknown error has occured while creating discord session.\n" +
```
Comment (mirror-kt) [inline setup/setup.go:17]: log.Panicf(formatString, args...)を使えばSprintfいらなそう

File: setup/setup.go [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+	discord.Token = utils.GetToken()
+
+	discord.AddHandler(handlers.OnMessageCreate)
+	discord.AddHandler(handlers.OnVoiceStateChanged)
+
+	if err = discord.Open(); err != nil {
+		displayError := fmt.Sprintf("Error: Unknown error has occured while opening discord session.\n" +
+			"Detail:%s\n", err)
+		log.Panicf(displayError)
```
Comment (mirror-kt) [inline setup/setup.go:30]: Panicfを使ってるのにSprintfでフォーマットしてる

**Review by mirror-kt — CHANGES_REQUESTED** [review]

File: setup/setup.go (hunk 8 above)
Comment (Colk-tech) [reply setup/setup.go:17]: This will be resolved in #2.

File: setup/setup.go (hunk 9 above)
Comment (Colk-tech) [reply setup/setup.go:30]: This will be resolved in #2.
